### PR TITLE
Add a default .m.rule.tombstone push rule

### PR DESCRIPTION
**Pull request by turt2live**
In support of MSC1930: https://github.com/matrix-org/matrix-doc/pull/1930
See also https://github.com/matrix-org/matrix-react-sdk/pull/2798

### CODE REVIEW

**Comment by turt2live** [comment]
The MSC has cleared FCP, which means this is ready for actual review.

**Comment by erikjohnston** [comment]
This won't cause clients to be told about this new push rule down sync, but OTOH presumably we don't want to bother spending the time wiring in that update functionality now?

**Comment by turt2live** [comment]
I'm not sure how possible it is to do the background update (at least at my level). I can probably stumble my way through getting the update to work, however I don't have a good metric for progress or point of continuance when processing users.

Theory was to figure out all the users which need the push rule and call `self.store.set_push_rule_enabled` over the set, which causes the `push_rules_stream` to work and for the values to be updated in the `push_rules_enabled` table. 

For progress I wanted to do some sort of lookup for who has push notifications enabled and who doesn't have an entry for `.m.rule.tombstone` in the table - the goal was to avoid updating push rules for the millions of bridge users which don't need it. However, trying to select who has the master rule disabled (because it is backwards) yields interesting results:
```
t2lsynapse=# select * from push_rules_enable where rule_id = 'global/override/.m.rule.master' and user_name = '@travis:t2l.io';
 id | user_name | rule_id | enabled
----+-----------+---------+---------
(0 rows)

t2lsynapse=# select count(*) from push_rules_enable where user_name = '@travis:t2l.io';
 count
-------
   593
(1 row)
```

I do have push notifications working though, and looking at another homeserver's database I can see that sometimes the push rule is there and sometimes not :(

So in summary: I think it's fine to expect clients to make their own defaults, or resync, or something for this rule because writing this background update looks painful. In practice I don't think there's very many clients which actually use push rules.
